File: app/src/main/java/com/vpaliy/mediaplayer/di/component/PlaybackComponent.java
package com.vpaliy.mediaplayer.di.component;

import com.vpaliy.mediaplayer.di.module.MediaPlayerModule;
import com.vpaliy.mediaplayer.domain.playback.PlaybackScope;
import com.vpaliy.mediaplayer.playback.MusicPlaybackService;

import dagger.Component;

@PlaybackScope
@Component(dependencies = ApplicationComponent.class,
    modules = MediaPlayerModule.class)
public interface PlaybackComponent {
    void inject(MusicPlaybackService service);
}

File: app/src/main/java/com/vpaliy/mediaplayer/di/component/PlaybackComponent.kt
package com.vpaliy.mediaplayer.di.component

import com.vpaliy.mediaplayer.di.module.MediaPlayerModule
import com.vpaliy.mediaplayer.playback.MusicPlaybackService
import com.vpaliy.mediaplayer.ui.player.PlayerActivity
import com.vpaliy.mediaplayer.domain.playback.PlaybackScope
import dagger.Component

@PlaybackScope
@Component(dependencies = [(ApplicationComponent::class)], modules = [(MediaPlayerModule::class)])
interface PlaybackComponent {
  fun inject(service: MusicPlaybackService)
  fun inject(activity: PlayerActivity)
}
